contracts/apps/disputable/IAgreement.sol

```diff
@@ -0,0 +1,101 @@
+/*
+ * SPDX-License-Identifier:    MIT
+ */
+
+pragma solidity ^0.4.24;
+
+import "../../acl/IACLOracle.sol";
+import "../../lib/token/ERC20.sol";
+import "../../lib/arbitration/IArbitrable.sol";
+
+
+contract IAgreement is IArbitrable, IACLOracle {
+
+    event Signed(address indexed signer, uint256 settingId);
+    event ActionSubmitted(uint256 indexed actionId);
+    event ActionClosed(uint256 indexed actionId);
+    event ActionChallenged(uint256 indexed actionId, uint256 indexed challengeId);
+    event ActionSettled(uint256 indexed actionId, uint256 indexed challengeId);
+    event ActionDisputed(uint256 indexed actionId, uint256 indexed challengeId);
+    event ActionAccepted(uint256 indexed actionId, uint256 indexed challengeId);
+    event ActionVoided(uint256 indexed actionId, uint256 indexed challengeId);
+    event ActionRejected(uint256 indexed actionId, uint256 indexed challengeId);
+    event DisputableAppRegistered(address indexed disputable);
+    event DisputableAppUnregistered(address indexed disputable);
+
+    enum ChallengeState {
+        Waiting,
+        Settled,
+        Disputed,
+        Rejected,
+        Accepted,
+        Voided
+    }
+
+    function sign() external;
+
+    function newAction(uint256 _disputableActionId, bytes _context, address _submitter) external returns (uint256);
+
+    function closeAction(uint256 _actionId) external;
+
+    function challengeAction(uint256 _actionId, uint256 _settlementOffer, bool _finishedSubmittingEvidence, bytes _context) external;
+
+    function settle(uint256 _actionId) external;
+
+    function disputeAction(uint256 _actionId, bool _finishedSubmittingEvidence) external;
+
+    function register(
+        address _disputable,
+        ERC20 _collateralToken,
+        uint256 _actionAmount,
+        uint256 _challengeAmount,
+        uint64 _challengeDuration
+    )
+        external;
+
+    function unregister(address _disputable) external;
+
+    function getSigner(address _signer) external view returns (uint256 lastSettingIdSigned, bool mustSign);
+
+    function getAction(uint256 _actionId) external view
+        returns (
+            address disputable,
+            uint256 disputableActionId,
+            uint256 collateralId,
```

Comment:
I prefer being explicit haha, let's go with `disputableCollateralRequirementId`

Reply:
Sounds good!

I was thinking about this statement I made:

> you might think the collateral settings live globally in the app, rather than per disputable

And realized that, even if it's a bit fictional, the current interface doesn't include anything that forces this arrangement, and so it's fine to be more general.